### DISCUSSION TITLE
Rate limit / usage issues

### DISCUSSION THREAD

**Discussion by hyyyper** [etsy/open-api#1125] [Q&A]
I am noticing some concerning behavior around rate-limit/usage.

Rate-limits no longer seem to apply per-user.
Also some of my logs are showing usage fluctuating from ~100% -> ~0% -> ~100%.

Anyone else experiencing strange rate-limit/usage issues?

An example from my logs for one user:
```
    "x-limit-per-day": "15000",
    "x-remaining-today": "432",
    "date": "Thu, 05 Oct 2023 23:10:33 GMT",
```

```
    "x-limit-per-day": "15000",
    "x-remaining-today": "14983",
    "date": "Thu, 05 Oct 2023 23:23:03 GMT",
```

```
    "x-limit-per-day": "15000",
    "x-remaining-today": "415",
    "date": "Thu, 05 Oct 2023 23:34:33 GMT",
```

**Comment by gaborp** [comment]
We started seeing strange behavior on X-Remaining-Today response header also.
We were never even near hitting the limit, yesterday we have noticed that we are getting 0 as available requests back. Strangely enough Etsy is responding with valid response, not by HTTP 429 as I would expect if we would actually burn all available requests.

We have also received negative number in response header: X-Remaining-Today = -1.

**Reply by hyyyper** [reply]
That's very strange indeed.

Good to know it's not just me :)

I contacted the developer support by email yesterday. I'll update here when I get a response.

**Comment by seandmccarthy** [comment]
We've been seeing some intermittent but regular flapping between the correct figure and a negative number. Feels like they have a server that hasn't sync'ed API call stats (or maybe it fails to read that data?) and you occasionally hit that server behind whatever load balancing they have. At worst we're seeing the alternating between bad and good roughly every second call. Of course their status page indicates everything is fine :)
